static/app/views/eventsV2/table/cellAction.tsx

```diff
@@ -430,21 +458,71 @@ class CellAction extends Component<Props, State> {
           </Reference>
           {menu}
         </Manager>
-      </MenuRoot>
+      </MenuActionButton>
+    );
+  }
+
+  renderContext() {
+    const {isContextPopperOpen: isOpen} = this.state;
+
+    if (!hasContext(this.props.dataRow, this.props.column)) {
+      return null;
+    }
+
+    const context = !isOpen
+      ? null
+      : createPortal(
+          <Popper placement="top" modifiers={this.modifiers}>
+            {({ref: popperRef, style, placement, arrowProps}) => (
+              <Menu
+                ref={ref => {
+                  (popperRef as Function)(ref);
+                  this.contextEl = ref;
+                }}
+                style={style}
+              >
+                <MenuArrow
+                  ref={arrowProps.ref}
+                  data-placement={placement}
+                  style={arrowProps.style}
+                />
+                <QuickContext dataRow={this.props.dataRow} column={this.props.column} />
+              </Menu>
+            )}
+          </Popper>,
+          this.portalEl
+        );
+
+    return (
+      <ContextActionButton>
+        <Manager>
+          <Reference>
+            {({ref}) => (
+              <MenuButton ref={ref} onClick={this.handleContextToggle}>
+                <IconShow size="sm" data-test-id="cell-action-context" color="subText" />
+              </MenuButton>
+            )}
+          </Reference>
+          {context}
+        </Manager>
+      </ContextActionButton>
     );
   }
 
   render() {
-    const {children} = this.props;
-    const {isHovering} = this.state;
+    const {children, organization} = this.props;
+    const {isHovering, isContextPopperOpen, isMenuPopperOpen} = this.state;
 
     return (
       <Container
         onMouseEnter={this.handleMouseEnter}
         onMouseLeave={this.handleMouseLeave}
       >
         {children}
-        {isHovering && this.renderMenu()}
+        <Feature organization={organization} features={['discover-quick-context']}>
```

Comment:
This will add the context menu to all the tables that use Cell Actions, we just want to show the quick context popover on the discover tables

Reply:
Hey, the "Wrapper with context button only around CellActions under TableView" a shot to tackle this.

Reply:
can we just add an explicit optional boolean flag that's a prop to this component like `showQuickContextMenu` and check for both that flag and the feature so we don't accidentally expose it (even if just internally) to the performance landing page for example as we add support for more fields.